src/huggingface_hub/hf_api.py

```diff
@@ -3625,10 +3633,13 @@ def update_repo_settings(
         # Build headers
         headers = self._build_hf_headers(token=token)
 
+        # Preparing the JSON payload for the PUT request
+        payload = {"gated": gated, "private": private}
```

Comment:
I would prepare the payload in 2 steps:

```py
payload: Dict = {}

if gated is not None:
    if gated not in ["auto", "manual", False]:
        raise ValueError(f"Invalid gated status, must be one of 'auto', 'manual', or False. Got '{gated}'.")
    payload["gated"] = gated

if private is not None:
    payload["private"] = private
```

The advantage of doing so is that adding a new parameter will just be a matter of adding a new `if` statement. Also, it makes it easier to log/print the payload value in case of debugging an issue. Doing a `{key: value for key, value in payload.items() if value is not None}` is not wrong per-se but makes it slightly less readable and less debuggable

Reply:
I was planning on a similar approach, but I opted for the dictionary comprehension `{key: value for key, value in payload.items() if value is not None}` to make the code more efficient.

Reply:
Sometime readibility / maintenance is more important than efficiency :)